### ISSUE TITLE
BUG - SEV 3 - All - Users told app isn't working when they're just not authorized for feature

### ISSUE THREAD

**Issue by mistymg** [department-of-veterans-affairs/va-mobile-app#7314]
## What happened?

When checking for what we show a user when they're not authorized for disability rating and claims, I noticed we're not showing the incorrect message. Karsyn doesn't have have a disability rating and not authorized for claims. But we're showing him the app-isn't-working red alert.

## Specs:

- Device: iPhone SE
- OS: iOS
- User Account: Karsyn
- Accessibility State: none

## Steps to Reproduce

- Sign in to mobile app using Karsyn's credentials (who doesn't have a DR)
- Tap Benefits in bottom navigation
- Tap Disability rating
- You get this:

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNG5oQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--e116ab4040d0a7effc3b6e40c2cb64ad043218ec/IMG-0182.PNG)

OR
- Sign in to mobile app using Karsyn's credentials (who doesn't have access to claims)
- Tap Benefits in bottom navigation
- Tap claims
- You get this:

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNDNoQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--ae592009dc60ddd3a4ee74710bb024ce538e373a/IMG-0183.PNG)

## Desired behavior

The message we show users should be accurate when they're not authorized for a feature.

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: Low 
- Frequency: Low 

- 3 - Low

## Screen shot(s) and additional information

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  

</details>

## Ticket Checklist
- [X] Steps to reproduce are defined
- [X] Desired behavior is added
- [X] Labels added (front-end, back-end, relevant feature, accessibility, etc)
- [ ] Estimate of 1 added (for front-end tickets)
- [ ] Added either to the relevant feature epic (if found during new feature implementation), or the [bug epic](https://app.zenhub.com/workspaces/va-mobile-product-view-610035bc5395bb000e62e529/issues/gh/department-of-veterans-affairs/va-mobile-app/5367)

**Comment by mistymg** [comment]
@bischoffa Another bug for the backlog!

**Comment by bischoffa** [comment]
@mistymg is there existing content that we show today for when they are unauthorized for a feature?

**Comment by mistymg** [comment]
I'm not sure without taking the time to dig in and see. I thought we had something so I'm surprised we're showing this. Eng might be able to answer this more quickly (or rather, the first place I'd start is asking them).

**Comment by mistymg** [comment]
This bug is not ready to be worked by Eng. UX and Content need to determine if the red alert box is what we should use and what content should be.

**Comment by mistymg** [comment]
This bug might now be ready to work by Eng.

Originally, I thought we didn't have a message when a user isn't authorized for claims/DR. But according to Dylan in [this thread](https://adhoc.slack.com/archives/C0212B872MT/p1702315047734969?thread_ts=1700243310.470829&cid=C0212B872MT), we do have this type of message. It's just not showing for Karysn (user not authorized for claims/DR).

**Comment by TKDickson** [comment]
@mistymg We only show that message if a user is not authorized for both claims and appeals. Karsyn is authorized for appeals (and returning an error, since he doesn't have any appeals) and also gets an error for claims (because he's not authorized) - hence the error message.

@dumathane (not a veteran) created an id.me account in production and it was returning "true" for `authorizedServices: appeals` (and appointments, although that one is less relevant to this ticket). [Pinged the backend team](https://adhoc.slack.com/archives/C0206QFMXFU/p1712953154197369) to try to figure out how we're determining folks are authorized for appeals, because that seems to be the true root cause of this issue.  Moving to blocked for now.

_Also figuring it out seems like it will directly relate to #7746, since the "authorized for appeals, but not claims, and getting an error for both" is the main reason (in staging) that we're seeing that home screen freeze._

**Comment by TKDickson** [comment]
The thread I linked above, is continued in [DSVA in this thread](https://dsva.slack.com/archives/C018V2JCWRJ/p1712954223621319?thread_ts=1712946688.274359&cid=C018V2JCWRJ). 

Relevant excerpts:

> this is the policy for appeals: user.loa3? && user.ssn.present?
> this is (effectively) the policy for appointments:
> user.icn.present? && user.loa3?

> For the mobile app, we force all credentials to be identity proofed, so user.loa3? should be true for all users who have successfully authenticated to the mobile app
> Additionally, if you are identity proofed and have logged in through a VA web system, an MPI record will be created for you. This happens whether you are a veteran or not. So user.icn should also be true for all users who have successfully authenticated to the mobile app
> Finally, in order to be identity proofed, at least on id.me and login.gov, I'm pretty sure SSN is a requirement, so user.ssn should also be true for all users on the mobile app

So essentially - everyone on the mobile app will always be authorized for appeals and appointments, with various downstream impacts, and we probably need to figure out if we want to make backend or frontend (or both) changes to tighten this up.

**Comment by TKDickson** [comment]
I have confirmed we are currently correctly showing the 'you don't have a disability rating', rather than the red app error screen, for karsyn (who is not authorized). Screenshot below. (Whether this is the BEST content for these users, I'm not sure - but it is better than 'the app isn't working' from the original writeup)

The other portion of this - what to do with users in the weird claims/appeals state - is a subset of/a bit duplicative with #8752. I'm going to close this one (because of the DR stuff being resolved) and use the other to track progress on claims/appeals.

Closing.

<img width="396" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMVFIQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--a6a87b7194bfbb32677ed9ce2d7013831c6363cf/image.png" alt="image.png" />
